### PR TITLE
forbid warnings in proof-producer

### PR DESCRIPTION
**Pull request by AndreyMlashkin**

### CODE REVIEW

File: crypto3/libs/blueprint/include/nil/blueprint/bbf/generic.hpp [hunk 1]
```diff
@@ -220,6 +220,8 @@ namespace nil {
                             }
                             BOOST_ASSERT(C == at.constant(get_col(col,t), get_row(row)));
                         break;
+                        default:
```
Comment (oclaw) [inline crypto3/libs/blueprint/include/nil/blueprint/bbf/generic.hpp:223]: unhandled enum case in switch

File: crypto3/libs/blueprint/include/nil/blueprint/zkevm/zkevm_word.hpp [hunk 2]
```diff
@@ -36,22 +36,20 @@
...
     namespace blueprint {
 
-        constexpr static const
-            boost::multiprecision::number<
+        constexpr inline boost::multiprecision::number<
                 boost::multiprecision::backends::cpp_int_modular_backend<257>> zkevm_modulus =
                         0x10000000000000000000000000000000000000000000000000000000000000000_cppui_modular257;
 
-        constexpr static const boost::multiprecision::backends::modular_params<
+        constexpr inline boost::multiprecision::backends::modular_params<
```
Comment (oclaw) [inline crypto3/libs/blueprint/include/nil/blueprint/zkevm/zkevm_word.hpp:43]: subobject linkage (potential ODR violation)

File: crypto3/libs/blueprint/include/nil/blueprint/zkevm_bbf/opcodes/mul.hpp [hunk 3]
```diff
@@ -110,13 +110,13 @@ namespace nil {
...
                     const opcode_input_type<FieldType, GenerationStage::ASSIGNMENT> &current_state
-                ) {
+                ) override {
                     zkevm_mul_bbf<FieldType, GenerationStage::ASSIGNMENT> bbf_obj(context, current_state);
                 }
                 virtual void fill_context(
                     typename generic_component<FieldType, GenerationStage::CONSTRAINTS>::context_type &context,
                     const opcode_input_type<FieldType, GenerationStage::CONSTRAINTS> &current_state
-                ) {
+                ) override {
```
Comment (oclaw) [inline crypto3/libs/blueprint/include/nil/blueprint/zkevm_bbf/opcodes/mul.hpp:119]: implicit override (lots of them)

**Comment by AndreyMlashkin** [comment]
Our first warning-free project!

**Review by martun — APPROVED** [review]

**Pull request merged**